src/Contractors.jl

```diff
@@ -48,192 +36,104 @@ derivative(f) = x-> f(Taylor1([x,1.],1))[1]
 #using DualNumbers
 #derivative(f) = x->f(Dual(x, 1..1)).epsilon
 
+
 """
-Compute a single root with (possibly multivariate) interval Newton
+Compute the guaranteed interval preimage f⁻¹(y) of a point y ∈ R according to the function f: [a,b] → R. 
+This preimage is guaranteed to lie inside the real interval `x = hull(a,b)`.
+
+`f` must be strictly monotonic on [a,b], at least in the sense of MonotonicBranch functions (allowing for uncertainty on the endpoints).
+`f` must be differentiable; zero and infinite derivative are allowed.
 
-x must be an Interval (univariate) or IntervalBox (multivariate)
+`y1`, `y2` must be guaranteed to be equal to f(a), f(b) or "more outside", so that  y ∉ hull(y1,f(z)) ⟹ f⁻¹(y) ∉ hull(a, z).
 
 Stops when the interval reaches a fixed point, when the diameter is smaller than ε,
 or when max_iter iterations are reached (with an error)
 """
-root(f, x; ϵ, max_iter) = root(f, derivative(f), x; ϵ, max_iter)
-
-function root(f::Function, f′::Function, x::Interval; ϵ, max_iter)
+preimage_monotonic(y, f::Function, x::Interval, (y1, y2); ϵ, max_iter) = preimage_monotonic(y, f::Function, derivative(f), x::Interval, (y1, y2); ϵ, max_iter)
```

Comment:
That's a good suggestion. I am also not sure what the better approach is. Another possible change in the future is that we can just implement `preimage_increasing`, since the decreasing branch case is treated by changing signs now. This would possibly save some checks, but it is just a minor thing.